workshops/forms.py

```diff
@@ -109,8 +116,92 @@ class DebriefForm(forms.Form):
     )
 
 
-# forms below are created using a factory instead of class
-# they're used on event edit page
-EventForm = modelform_factory(Event, fields="__all__")
-TaskForm = modelform_factory(Task, fields="__all__",
-                             widgets={'event': HiddenInput})
+class EventForm(forms.ModelForm):
+
+    site = selectable.AutoCompleteSelectField(
+        lookup_class=lookups.SiteLookup,
+        label='Site',
+        required=True,
+        help_text=AUTOCOMPLETE_HELP_TEXT
+    )
+
+    organizer = selectable.AutoCompleteSelectField(
+        lookup_class=lookups.SiteLookup,
+        label='Organizer',
+        required=True,
+        help_text=AUTOCOMPLETE_HELP_TEXT
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['start'].help_text = DATE_HELP_TEXT
+        self.fields['end'].help_text = DATE_HELP_TEXT
+
+    def clean_slug(self):
+        # Ensure slug is not an integer value for Event.get_by_ident
+        data = self.cleaned_data['slug']
+
+        try:
+            nonInt = int(data)
+            raise forms.ValidationError("Slug must not be an integer-value.")
+        except ValueError:
+            pass
+
+        return data
+
+    class Meta:
+        model = Event
+        fields = '__all__'
+
+
+class TaskForm(forms.ModelForm):
+
+    person = selectable.AutoCompleteSelectField(
+        lookup_class=lookups.PersonLookup,
+        label='Person',
+        required=True,
+        help_text=AUTOCOMPLETE_HELP_TEXT
+    )
+
+    def __init__(self, *args, **kwargs):
+        event = kwargs.pop('event', None)
+        super().__init__(*args, **kwargs)
+        if event:
+            self.instance.event = event
+
+    class Meta:
+        model = Task
+        exclude = ('event', )
+
+
+class TaskFullForm(TaskForm):
+
+    event = selectable.AutoCompleteSelectField(
+        lookup_class=lookups.EventLookup,
+        label='Event',
+        required=True,
+        help_text=AUTOCOMPLETE_HELP_TEXT
+    )
+
+    class Meta:
+        model = Task
+        fields = '__all__'
+
+
+class PersonForm(forms.ModelForm):
+
+    airport = selectable.AutoCompleteSelectField(
+        lookup_class=lookups.AirportLookup,
+        label='Airport',
+        required=False,
+        help_text=AUTOCOMPLETE_HELP_TEXT
+    )
```

Comment:
[AutoComboboxSelectWidget](http://django-selectable.readthedocs.org/en/latest/widgets.html#autocomboboxselectwidget) for `airport`.